app/models/admin_user.rb

```diff
@@ -52,6 +56,23 @@ def renew_otp_secret_key!
     save!
   end
 
+  def active_tags
+    admin_user_taggings.pluck(:tag_id)
+  end
+
+  def can_manage_tag?(tag)
+    AdminUser.where(%{not exists (select id from admin_user_taggings where admin_user_id = ?) 
+      or exists (select id from admin_user_taggings where admin_user_id = ? and tag_id = ?)}, self.id, self.id, tag.id)
+      .exists?
```

Comment:
Creo que acá podemos sacrificar velocidad por legibilidad.
```suggestion
    !admin_user_taggings.exists? ||
      admin_user_taggings.where(tag: tag).exists?
```